### ISSUE TITLE
Modify AuthUI to support linking accounts

### ISSUE THREAD

**Issue by fungc-io** [authgear/authgear-server#4109]
![image.png](https://uploads.linear.app/860e83ae-05f0-4c4b-bd93-8a4b8d251836/7a6c1b0a-2fa3-47f4-b911-ace2726eb2d7/3994d7f7-ad6a-40bf-bcc2-f7aa02282ed2?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzg2MGU4M2FlLTA1ZjAtNGM0Yi1iZDkzLThhNGI4ZDI1MTgzNi83YTZjMWIwYS0yZmEzLTQ3ZjQtYjkxMS1hY2UyNzI2ZWIyZDcvMzk5NGQ3ZjctYWQ2YS00MGJmLWJjYzItZjdhYTAyMjgyZWQyIiwiaWF0IjoxNzEyNjgyMjcxLCJleHAiOjMzMjgzMjQyMjcxfQ.qdr3QGENljgwJEsOSKp2enzqAminFIKC1ILrPAcdyQ0 "=786x")

Figma: [https://www.figma.com/file/LRE8EfCaTfYxC0lk5AeMmE/New-Auth-UI?type=design&node-id=2365-15938&mode=design&t=LXscrPWftY0rV2Wc-4](https://www.figma.com/file/LRE8EfCaTfYxC0lk5AeMmE/New-Auth-UI?type=design&node-id=2365-15938&mode=design&t=LXscrPWftY0rV2Wc-4)

---

- [ ] Add `prompt=login` to provider

**Comment by fungc-io** [comment]
@tung2744 @louischan @newman see if the design above works?

**Comment by fungc-io** [comment]
@tung2744 @joyz i've updated the design based on our discussion today. Any thoughts?:

<img src="https://uploads.linear.app/860e83ae-05f0-4c4b-bd93-8a4b8d251836/7a4b1e35-01cd-4042-ab02-8da615de7989/02cbbab8-8d47-49e3-8870-ea7528403bd8?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzg2MGU4M2FlLTA1ZjAtNGM0Yi1iZDkzLThhNGI4ZDI1MTgzNi83YTRiMWUzNS0wMWNkLTQwNDItYWIwMi04ZGE2MTVkZTc5ODkvMDJjYmJhYjgtOGQ0Ny00OWUzLTg4NzAtZWE3NTI4NDAzYmQ4IiwiaWF0IjoxNzEyNzU4NDc5LCJleHAiOjMzMjgzMzE4NDc5fQ.lKlsb2Y_vcFHCJJ-rq0RqOSLiLNZh88TJYuznIJhiig " alt="image.png" width="786" height="1688" />

**Comment by tung2744** [comment]
<img src="https://uploads.linear.app/860e83ae-05f0-4c4b-bd93-8a4b8d251836/a964c01e-7775-4804-947f-3e370688e422/9f5ef0a5-d0c0-4fb4-973b-9928e250e657?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzg2MGU4M2FlLTA1ZjAtNGM0Yi1iZDkzLThhNGI4ZDI1MTgzNi9hOTY0YzAxZS03Nzc1LTQ4MDQtOTQ3Zi0zZTM3MDY4OGU0MjIvOWY1ZWYwYTUtZDBjMC00ZmI0LTk3M2ItOTkyOGUyNTBlNjU3IiwiaWF0IjoxNzEzOTU5Njk0LCJleHAiOjMzMjg0NTE5Njk0fQ.gdzQCug-SfzoULYjtChqIwXyk_Z7D2qPH10KODydQh8 " alt="Screenshot 2024-04-24 at 7.53.06 PM.png" width="429" height="386" />

@fungc-io There is a case that the user signed in to a incorrect oauth account after choosing sign in with OAuth. I've created this error message to handle the error, please let me know if any update is needed.

**Comment by fungc-io** [comment]
Ok 👌

**Comment by tung2744** [comment]
@fungc-io @louischan Sorry I've found one more problem in this issue. Now the config specify the "action" per linking, which means, the "action" of each conflicted identities could be different if they are conflicted by different linking config. So maybe the message of each button should be different.<br>As now we actually haven't support login without link, maybe for simplicity we always show "Log in to an existing account to link it", and revise the design later if we supported login without link?
